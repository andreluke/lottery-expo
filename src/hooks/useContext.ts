import { useContext } from 'react';
import { LotteryContext } from '../contexts/LotteryContexts';

export const useLottery = () => {
  return useContext(LotteryContext);
};
