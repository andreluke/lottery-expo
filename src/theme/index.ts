// src/themes/theme.ts
import { StyleSheet } from "react-native";

export const theme = {
  colors: {
    background: "#ffffff",
    menuBackground: '#f4fcff',
    text: "#333333",
    megaSena: {
      background: "#209869",
      text: "#ffffff",
    },
    quina: {
      background: "#260085",
      text: "#ffffff",
    },
    timemania: {
      background: "#FFF600",
      text: "#049645",
    },
    loading: "#209869",
  },
  fontSizes: {
    title: 24,
    subtitle: 14,
    text: 16,
  },
  spacing: {
    small: 4,
    medium: 8,
    large: 16,
  },
  button: '#2d74da'
};

export const styles = StyleSheet.create({
  ballContainer: {
    flexDirection: "row",
  },
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  ball: {
    borderRadius: 25,
    width: 50,
    height: 50,
    justifyContent: "center",
    alignItems: "center",
    margin: theme.spacing.small,
  },
  number: {
    fontSize: theme.fontSizes.text,
    fontWeight: "bold",
  },
  subText: {
    fontSize: theme.fontSizes.subtitle,
    color: "#5d5d5d",
    marginTop: theme.spacing.small,
  },
  title: {
    fontSize: theme.fontSizes.title,
    alignSelf: "center",
    marginBottom: theme.spacing.large,
    fontWeight: "bold",
  },
});

export const style = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.colors.menuBackground,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 20,
    color: theme.colors.text,
  },
  buttonText: {
    color: theme.colors.background,
    fontSize: 16,
},
button: {
  backgroundColor: theme.button,
  paddingVertical: 10,
  borderRadius: 35,
  alignItems: 'center',
  width: '40%',
},
});