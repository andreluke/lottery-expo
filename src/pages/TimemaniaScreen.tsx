import React from 'react';
import { View, Text, ActivityIndicator } from 'react-native';
import { useLottery } from '../hooks/useContext';
import Ball from '../components/Ball';
import { theme } from '../theme';
import { styles } from '../theme';

const TimemaniaScreen: React.FC = () => {
  const { results, loading } = useLottery();

  // if (loading) {
  //   return <ActivityIndicator size="large" color={theme.colors.timemania.background} />;
  // }

  if (loading) {
    return <Text style={{fontSize: theme.fontSizes.text}}>Carregando...</Text>;
  }

  const timemaniaResults = results?.timemania;

  return (
    <View style={styles.container}>
      <Text style={{fontSize: theme.fontSizes.title, alignSelf: "center", marginBottom: theme.spacing.large, fontWeight: "bold"}}>TIMEMANIA</Text>
    <View style={styles.ballContainer}>
    {timemaniaResults?.dezenas.map((dezena: number) => (
        <Ball key={dezena} number={dezena} color={theme.colors.timemania.background} text={theme.colors.timemania.text} spacing={theme.spacing.small}/>
      ))}
    </View>
    <Text style={{fontSize: theme.fontSizes.subtitle, color: "#5d5d5d", marginTop: theme.spacing.small}}> {timemaniaResults.dataPorExtenso}</Text>
    </View>
  );
};

export default TimemaniaScreen;
