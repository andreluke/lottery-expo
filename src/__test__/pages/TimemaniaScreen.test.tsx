import React from 'react';
import renderer from 'react-test-renderer';
import TimemaniaScreen from '../../pages/TimemaniaScreen';
import { useLottery } from '../../hooks/useContext';
import { styles } from '../../theme';
import { Text } from 'react-native';

// Mock do hook useLottery
jest.mock('../../hooks/useContext', () => ({
  useLottery: jest.fn(),
}));

describe('TimemaniaScreen', () => {
  it('deve exibir o estado de carregamento corretamente', () => {
    // Mock para simular o estado de carregamento
    (useLottery as jest.Mock).mockReturnValue({
      results: null,
      loading: true,
    });

    const tree = renderer.create(<TimemaniaScreen />).toJSON();
    
    // Verifica se o snapshot do estado de carregamento está correto
    expect(tree).toMatchSnapshot();
  });

  it('deve exibir o título do Timemania', () => {
    // Mock para simular resultados de Mega-Sena com o estado de carregamento falso
    (useLottery as jest.Mock).mockReturnValue({
      results: {
        timemania: {
          dezenas: [9, 14, 55, 77, 2, 66, 38],
          dataPorExtenso: '29 de Outubro de 2024',
        },
      },
      loading: false,
    });

    const tree = renderer.create(<TimemaniaScreen />).toJSON();
    
    // Verifica se o snapshot do componente com resultados da Mega-Sena está correto
    expect(tree).toMatchSnapshot();
  });

  it('deve exibir os números corretos da loteria', () => {
    // Mock para simular resultados da Mega-Sena
    (useLottery as jest.Mock).mockReturnValue({
      results: {
        timemania: {
          dezenas: [9, 14, 55, 77, 2, 66, 38],
          dataPorExtenso: '29 de Outubro de 2024',
        },
      },
      loading: false,
    });

    const component = renderer.create(<TimemaniaScreen />);
    const instance = component.root;

    // Verifica se os números corretos são exibidos no componente
    const ballNumbers = instance.findAllByType(Text).map((text) => text.props.children);
    
    expect(ballNumbers).toContain(9);
    expect(ballNumbers).toContain(14);
    expect(ballNumbers).toContain(55);
    expect(ballNumbers).toContain(77);
    expect(ballNumbers).toContain(2);
    expect(ballNumbers).toContain(66);
    expect(ballNumbers).toContain(38);
  });

  it('deve exibir a data correta do sorteio', () => {
    // Mock para simular o retorno de resultados da Mega-Sena com data
    (useLottery as jest.Mock).mockReturnValue({
      results: {
        timemania: {
          dezenas: [9, 14, 55, 77, 2, 66, 38],
          dataPorExtenso: '29 de Outubro de 2024',
        },
      },
      loading: false,
    });

    const component = renderer.create(<TimemaniaScreen />);
    const instance = component.root;

    const dateTextArray = instance.findByProps({ style: styles.subText }).props.children;
    
    // Corrige o teste unindo os elementos do array em uma string e removendo espaços adicionais
    const dateText = dateTextArray.join('').trim();

    expect(dateText).toBe('29 de Outubro de 2024');
  });
});
