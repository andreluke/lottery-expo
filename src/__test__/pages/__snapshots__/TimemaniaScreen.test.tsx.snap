// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimemaniaScreen deve exibir o estado de carregamento corretamente 1`] = `
<Text
  style={
    {
      "fontSize": 16,
    }
  }
>
  Carregando...
</Text>
`;

exports[`TimemaniaScreen deve exibir o t√≠tulo do Timemania 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 16,
      }
    }
  >
    TIMEMANIA
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#FFF600",
            "margin": 4,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#049645",
            },
          ]
        }
      >
        9
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#FFF600",
            "margin": 4,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#049645",
            },
          ]
        }
      >
        14
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#FFF600",
            "margin": 4,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#049645",
            },
          ]
        }
      >
        55
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#FFF600",
            "margin": 4,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#049645",
            },
          ]
        }
      >
        77
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#FFF600",
            "margin": 4,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#049645",
            },
          ]
        }
      >
        2
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#FFF600",
            "margin": 4,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#049645",
            },
          ]
        }
      >
        66
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#FFF600",
            "margin": 4,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#049645",
            },
          ]
        }
      >
        38
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#5d5d5d",
        "fontSize": 14,
        "marginTop": 4,
      }
    }
  >
     
    29 de Outubro de 2024
  </Text>
</View>
`;
